mitodl/mit-open
# enforce import hierarchy by linting; finish removing old code

# What are the relevant tickets?
Closes #179 

# Description (What does it do?)
This PR enforces the import hierarchy described in https://github.com/mitodl/mit-open/blob/main/docs/architecture/front-end-component-structure.md via linting rules.

# How can this be tested?
1. Open any file in `frontends/mit-open/src/components/` directory and try to import something from `services`, `page-components`, or `pages. You should get an error with a meaningful error message.
    - Try both the typescript alias `@/pages/X/X` version of the import and the relative version, e.g., `../../pages/X/X`
2. Try similar things in other directories, as described in https://github.com/mitodl/mit-open/blob/main/docs/architecture/front-end-component-structure.md
